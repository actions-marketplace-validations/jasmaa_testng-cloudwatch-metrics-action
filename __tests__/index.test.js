const core = require('@actions/core');
const { run } = require('../src/index');

jest.mock('@actions/core');

const mockPutMetricData = jest.fn(() => {
  return {
    promise: jest.fn(),
  };
});

jest.mock('aws-sdk', () => {
  return {
    CloudWatch: jest.fn(() => {
      return {
        putMetricData: mockPutMetricData,
      };
    }),
    config: {
      update: jest.fn(),
    }
  };
});

describe('Publish TestNG results', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  })

  test.each([
    ['__tests__/basic_noTests'],
    ['__tests__/basic_onePass'],
    ['__tests__/basic_oneFail'],
    ['__tests__/basic_oneSkip'],
    ['__tests__/basic_oneUnknown'],
    ['__tests__/complex_multipleMethods'],
    ['__tests__/complex_multipleClasses'],
    ['__tests__/complex_multipleTests'],
    ['__tests__/complex_real1'],
  ])('test single suite for reports-path=%p', async (reportsPath) => {
    core.getInput = jest.fn().mockImplementation((name) => {
      return {
        'reports-path': reportsPath,
        'namespace': 'TestNG Canary',
      }[name];
    });

    await run();

    expect(mockPutMetricData).toHaveBeenCalledTimes(1);
    expect(mockPutMetricData.mock.calls[0][0]).toMatchSnapshot();
  });

  test('test multiple suites', async () => {
    core.getInput = jest.fn().mockImplementation((name) => {
      return {
        'reports-path': '__tests__/complex_multipleSuites',
        'namespace': 'TestNG Canary',
      }[name];
    });

    await run();

    expect(mockPutMetricData).toHaveBeenCalledTimes(2);
  });
});