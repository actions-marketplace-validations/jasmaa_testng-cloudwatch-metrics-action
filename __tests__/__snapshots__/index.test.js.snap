// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish TestNG results test single suite for reports-path="__tests__/basic_noTests" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/basic_oneFail" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/basic_onePass" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/basic_oneSkip" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/basic_oneUnknown" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 1,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/complex_multipleClasses" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 8,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 5,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 2,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/complex_multipleMethods" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 4,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 1,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/complex_multipleTests" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 2,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 0,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;

exports[`Publish TestNG results test single suite for reports-path="__tests__/complex_real1" 1`] = `
{
  "MetricData": [
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Total",
      "Unit": "Count",
      "Value": 4,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Passed",
      "Unit": "Count",
      "Value": 2,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Failed",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Skipped",
      "Unit": "Count",
      "Value": 1,
    },
    {
      "Dimensions": [
        {
          "Name": "TestSuite",
          "Value": "Surefire suite",
        },
      ],
      "MetricName": "Unknown",
      "Unit": "Count",
      "Value": 0,
    },
  ],
  "Namespace": "TestNG Canary",
}
`;
